### PR TITLE
Add index after adding data without vector schema for vector predicate

### PR DESCRIPTION
**Pull request by shivaji-kharse**

### CODE REVIEW

File: tok/hnsw/helper.go [hunk 1]
```diff
@@ -378,9 +382,15 @@ func (ph *persistentHNSW[T]) getVecFromUid(uid uint64, c index.CacheType, vec *[
 		return err
 	}
 	if data != nil {
-		index.BytesAsFloatArray(data.([]byte), vec, ph.floatBits)
+		inVec, err := types.ParseVFloat(string(data.([]uint8)))
```
Comment (harshil-goel) [inline tok/hnsw/helper.go:385]: We don't want to call ParseVFloat here. It's very slow

File: posting/index.go [hunk 2]
```diff
@@ -187,6 +187,31 @@ func (txn *Txn) addIndexMutations(ctx context.Context, info *indexMutationInfo)
 			}
 			return pbEdges, nil
 		}
+
+		if info.op != pb.DirectedEdge_DEL &&
+			len(data) > 0 &&
+			len(info.factorySpecs) > 0 {
+			// how we can know is this is a schema mutation
+			inVec, err := types.ParseVFloat(string(data[0].Value.([]uint8)))
```
Comment (harshil-goel) [inline posting/index.go:195]: Why are you doing this twice

File: posting/index.go [hunk 3]
```diff
@@ -1387,6 +1386,37 @@ func rebuildTokIndex(ctx context.Context, rb *IndexRebuild) error {
...
+					return []*pb.DirectedEdge{}, err
+				}
+
+				if err := p.addMutation(ctx, txn, &edge); err != nil {
+					return []*pb.DirectedEdge{}, err
+				}
+			}
+		}
+
 		err := pl.Iterate(txn.StartTs, 0, func(p *pb.Posting) error {
```
Comment (harshil-goel) [inline posting/index.go:1420]: No need to iterate when you already have the value in val. Just run the function inside iterate

**Review by harshil-goel — APPROVED** [review]

File: posting/index.go [hunk 4]
```diff
@@ -1387,39 +1386,86 @@ func rebuildTokIndex(ctx context.Context, rb *IndexRebuild) error {
...
+				Value: p.Value,
+				Tid:   types.TypeID(p.ValType),
+			}
+			edge.Lang = string(p.LangTag)
+
+			newEdges, err := processAddIndexMutation(&edge, val)
+			if err != nil {
+				return err
+			}
+			if !runForVectors {
```
Comment (harshil-goel) [inline posting/index.go:1457]: remove if

File: posting/index.go [hunk 5]
```diff
@@ -1387,39 +1386,86 @@ func rebuildTokIndex(ctx context.Context, rb *IndexRebuild) error {
...
+				p, err := txn.Get(inKey)
+				if err != nil {
+					return []*pb.DirectedEdge{}, err
+				}
+
+				if err := p.addMutation(ctx, txn, &edge); err != nil {
+					return []*pb.DirectedEdge{}, err
+				}
+
+				_, err = processAddIndexMutation(&edge, val)
```
Comment (harshil-goel) [inline posting/index.go:1436]: this should be after the if condition. it should get executed for both when val.td == types.vfloatid.

Comment (harshil-goel) [reply posting/index.go:1436]: write a test that fails because of this bug right now, then fix the bug

Comment (shivaji-kharse) [reply posting/index.go]: TestVectorIndexDropPredicate and TestVectorIndexWithoutSchema have already been added, and they fail on the main branch

**Review by harshil-goel — APPROVED** [review]

**Pull request merged**